src/opera/compare/template_comparer.py

```diff
@@ -0,0 +1,137 @@
+                                 comparisons, context)
+
+    def compare_node(self, node1, node2, context):
+        comparisons = {
+            "types": Comparison(self.compare_types),
+            "properties": MapComparison(self.compare_value),
+            "attributes": MapComparison(self.compare_none),
+            "requirements": ListComparison(self.compare_requirement,
+                                           self.get_name),
+            "capabilities": ListComparison(self.compare_capability,
+                                           self.get_name),
+            "interfaces": MapComparison(self.compare_interface),
+        }
+        new_context = TemplateContext(node1, node2,
+                                      context.workdir1, context.workdir2)
+        return self.compare_item(node1, node2, comparisons, new_context)
+
+    def compare_requirement(self, req1, req2, context):
+        comparisons = {
+            "target": Comparison(self.compare_target),
+            "relationship": Comparison(self.compare_relationship)
+        }
+        return self.compare_item(req1, req2, comparisons, context)
+
+    def compare_capability(self, cabability1, cabability2, context):
+        comparisons = {
+            "properties": MapComparison(self.compare_capability_property),
+            "attributes": MapComparison(self.compare_none)
+        }
+        return self.compare_item(cabability1, cabability2, comparisons,
+                                 context)
+
+    def compare_relationship(self, rel1, rel2, context):
+        comparisons = {
+            "types": Comparison(self.compare_types),
+            "properties": MapComparison(self.compare_value),
+            "attributes": MapComparison(self.compare_none),
+            "interfaces": MapComparison(self.compare_interface)
+        }
+        new_context = TemplateContext(rel1, rel2,
+                                      context.workdir1, context.workdir2)
+        return self.compare_item(rel1, rel2,
+                                 comparisons,
+                                 new_context)
+
+    def compare_interface(self, interface1, interface2, context):
+        comparisons = {
+            "operations": MapComparison(self.compare_operation)
+        }
+        return self.compare_item(interface1, interface2, comparisons, context)
+
+    def compare_operation(self, operation1, operation2, context):
+        comparisons = {
+            "primary": Comparison(self.compare_file),
+            "inputs": MapComparison(self.compare_value),
+            "dependencies": ListComparison(self.compare_file,
+                                           self.get_filename),
+            "artifacts": ListComparison(self.compare_file,
+                                        self.get_filename)
+        }
+        return self.compare_item(operation1, operation2, comparisons, context)
+
+    def compare_value(self, val1, val2, context):
```

Comment:
Add `@staticmethod` to all the methods that can be static and start methods that are used only within classes with underscores (_). Although Python does not have strong distinctions between “private” and “public” variables, you can still use underscores to indicate that some functions/variables are not really meant to be a part of the public interface of some class.

Reply:
I'm not a great fan of static methods as dealing with concurrency becomes less intuitive. Anyway the case I would prefer  instance methods in this case that I would like to provide an ability to override some comparer methods later on if needed, the same way it can be done with Storage class. I do not know whether making some of methods static and some not makes any sense.

Reply:
Ok, I agree that we can skip those `@staticmethod` annotations here. However, do take a look if you have any methods that are meant to be used just within certain classes and prefix them with underscores so that users will know that these methods shouldn't be touched ([example](https://github.com/xlab-si/xopera-opera/blob/master/src/opera/parser/tosca/csar.py#L155)).

Reply:
Done